plugins/komodo-plugin-service/src/main/java/org/komodo/osgi/PluginService.java

```diff
@@ -294,6 +294,8 @@ private void createDirectory(String path) {
     }
 
     private Bundle findBundleBySymbolicName(String symbolicName) throws Exception {
+        ArgCheck.isNotNull(framework, "framework");
```

Comment:
I can understand the logic but have 2 reservations:
1) assert statements do not tend to actually do anything unless the -ea switch is used. (see https://docs.oracle.com/javase/7/docs/technotes/guides/language/assert.html)
2) Its possible for
 * the PluginService to be instantiated (framework is inited)
 * the PluginService is shutdown (framework is set to null)
 * the PluginService is not started but another plublic method is called

So thinking about this, its probably better to modify this to throw an exception for those methods if the service is shutdown.